### PR TITLE
refactor(common): simplify date pipe metadata

### PR DESCRIPTION
**Pull request by kabrunko-dev**
Simplify `DatePipe` metadata properties.

### CODE REVIEW

File: packages/common/src/pipes/date_pipe.ts [hunk 1]
```diff
@@ -217,7 +217,6 @@ export const DATE_PIPE_DEFAULT_OPTIONS =
 // clang-format on
 @Pipe({
   name: 'date',
-  pure: true,
```
Comment (JeanMeche) [inline packages/common/src/pipes/date_pipe.ts:220]: Could you do the same for `I18nPluralPipe` & `I18nSelectPipe` ?

Comment (kabrunko-dev) [reply packages/common/src/pipes/date_pipe.ts:220]: Sure!

**Review by AndrewKushnir — APPROVED** [review]

**Review by JeanMeche — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
@kabrunko-dev thanks for creating this PR. Could you please update commit messages to include descriptions as described [here](https://github.com/angular/angular/actions/runs/7543355668/job/20910917632?pr=53939)?

**Comment by kabrunko-dev** [comment]
> @kabrunko-dev thanks for creating this PR. Could you please update commit messages to include descriptions as described [here](https://github.com/angular/angular/actions/runs/7543355668/job/20910917632?pr=53939)?

Thank you for letting me help!

**Comment by AndrewKushnir** [comment]
@kabrunko-dev I've noticed that you've updated one of two commits. Could you please update the second one as well? Thank you.

**Comment by kabrunko-dev** [comment]
> @kabrunko-dev I've noticed that you've updated one of two commits. [...]

@AndrewKushnir My fault

> [...] Could you please update the second one as well? Thank you.

It's done! Thank you.

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:602480561:BASE:602480585:1706563043090:782ab266).

**Review by AndrewKushnir — APPROVED** [review]

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit f3567bbc26644281461aa819d0986b1db0282b69.

**Pull request closed without merging**